observatory-platform/observatory/platform/utils/test_utils.py

```diff
@@ -302,12 +326,16 @@ def add_dataset(self, prefix: str = "") -> str:
 
         The BigQuery dataset will be deleted when the Observatory environment is closed.
 
-        :param prefix: an optional prefix for the dataset.
+        :param prefix: an optional additional prefix for the dataset.
         :return: the BigQuery dataset identifier.
         """
 
         self.assert_gcp_dependencies()
-        if prefix != "":
+        if (self.prefix != "") and (prefix != ""):
+            dataset_id = f"{self.prefix}_{prefix}_{random_id()}"
+        elif self.prefix != "" and (prefix == ""):
+            dataset_id = f"{self.prefix}_{random_id()}"
+        elif (self.prefix == "") and (prefix != ""):
             dataset_id = f"{prefix}_{random_id()}"
         else:
             dataset_id = random_id()
```

Comment:
This is an alternative way to make the prefix which might be easier to follow (an empty string evaluates to False as does None):

```python
parts = []
if self.prefix:
  parts.append(self.prefix)
if prefix:
  parts.append(prefix)
parts.append(random_id())
bucket_name = "_".join(parts)
```

You would also want to change the prefix parameter to `prefix: Optional[str] = None`